lib/cocoapods-core/specification/linter.rb

```diff
@@ -46,7 +46,8 @@ def initialize(spec_or_path)
       def lint
         @results = Results.new
         if spec
-          check_required_root_attributes
+          check_required_attributes
+          validate_name
```

Comment:
couldn't this just be `_validate_name`, and then wouldn't it be picked up automatically?